assimp/5.1.x: Unbundle pugixml.

sh0 opened the pull request:
Specify library name and version:  **assimp/1.0**

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

SpaceIm commented on recipes/assimp/5.x/conanfile.py:
```diff
@@ -145,6 +145,7 @@ def requirements(self):
 
         self.requires("minizip/1.2.11")
         self.requires("utfcpp/3.1.2")
+        self.requires("pugixml/1.11")
```
It's not a dependency of assimp < 5.1.0

sh0 replied:
Thanks for spotting that, fixed. Didn't see your comment edit at first so ended up with two commits.

SpaceIm commented:
Could you also ensure that vendored pugixml is not used by removing its folder?

```python
        # Take care to not use these vendored libs
        vendors = ["poly2tri", "rapidjson", "unzip", "utf8cpp", "zip"]
        if tools.Version(self.version) < "5.1.0":
            vendors.append("irrXML")
        else:
            vendors.append("pugixml")
        for vendor in vendors:
            tools.rmdir(os.path.join(self._source_subfolder, "contrib", vendor))
```

And by the way, draco is an optional dependency of gltf in assimp since 5.1.0 (disabled by default).

sh0 commented:
@SpaceIm I finally had time to work on this PR. I removed the vendored `pugixml` (and `draco`) library in contrib.

Also added support for draco library. Right now draco dependency is pulled in whenever gltf is enabled. Theoretically we can make it optional by adding a new `with_draco` flag. Would that be needed as well?

SpaceIm reviewed (approved):
> Theoretically we can make it optional by adding a new with_draco flag. Would that be needed as well?

It could be an improvement yes.

prince-chrismc commented on recipes/assimp/5.x/conanfile.py:
```diff
@@ -128,6 +128,12 @@ def _depends_on_poly2tri(self):
 
+    @property
+    def _depends_on_draco(self):
+        if tools.Version(self.version) < tools.Version("5.1.0"):
```
Is the double version required? I've never see that before 🤔

sh0 replied:
Ah, yes, it is not needed indeed, but should work fine. I probably copied it from somewhere without thinking, my bad.
